src/lib/flow/include/sol-flow-packet.h

```diff
@@ -135,6 +136,11 @@ struct sol_flow_packet *sol_flow_packet_new_direction_vector_components(double x
 int sol_flow_packet_get_direction_vector(const struct sol_flow_packet *packet, struct sol_direction_vector *direction_vector);
 int sol_flow_packet_get_direction_vector_components(const struct sol_flow_packet *packet, double *x, double *y, double *z);
 
+struct sol_flow_packet *sol_flow_packet_new_location(const struct sol_location *location);
+struct sol_flow_packet *sol_flow_packet_new_location_latlon(double lat, double lon);
```

Comment:
I'd say you can move this (and likely the other packet constructors) to take the structure, not the pointer, and use those macros or inline functions to create the structure from the numbers. So:

```
sol_flow_packet_new_location(const struct sol_location location)
sol_flow_packet_new_location(*my_ptr_to_location);
sol_flow_packet_new_location(SOL_LOCATION(lat, lon));
```